src/components/Widgets/SelectControl.js

```diff
@@ -15,12 +15,15 @@ export default class SelectControl extends React.Component {
       return <div>Error rendering select control for {field.get('name')}: No options</div>;
     }
 
-    const options = fieldOptions.map((option) => {
-      if (typeof option === 'string') {
-        return { label: option, value: option };
-      }
-      return option;
-    });
+    const options = [
+      ...(field.get('default', false) ? [] : [{ label: '<not set>', value: '' }]),
```

Comment:
I don't think the `<not set>` field should be a regular option -- I don't think it should be selectable. ~~After an option has been selected, the `<not set>` field should be hidden.~~

Reply:
I would change `{ label: '<not set>', value: '' }` to `{ label: '<not set>', disabled: true }`.

Reply:
@tech4him1 good point, I'll make the change.

Reply:
@Benaiah actually, now that you say that I'm not sure it does. If I use this PR like it is right now, it won't let me save if I select `<not set>`.

Reply:
@tech4him1 "not sure it does" :point_left: not sure it does what?

Reply:
@Benaiah There is no way, AFAIK, to save a select with an empty value unless `required: false`. Is that what you meant?

Reply:
@tech4him1 I meant that if you want to save an empty string, you can always add an option like `{ label: "Empty", value: "" }` to the config. `<not set>` shouldn't be selectable, as you suggested.

Reply:
Agreed.